tests/exploits/cve_2014_2120_test.py

```diff
@@ -0,0 +1,112 @@
+"""Unit tests for Agent Asteroid: CVE-2014-2120"""
+
+import requests
+import requests_mock as req_mock
+
+from agent import definitions
+from agent.exploits import cve_2014_2120
+
+
+# Mock response content representing a vulnerable and a secure page for the XSS test
+def create_mock_vulnerable_response() -> str:
+    """Create a mock HTML response with the XSS vulnerability reflected."""
+    return """
+    <html>
+        <body>
+            <input type="hidden" name="auth_handle" value="test123"'">
+            <input type="hidden" name="username" value="test123"'">
+        </body>
+    </html>
+    """
+
+
+def create_mock_secure_response() -> str:
+    """Create a mock HTML response without the XSS vulnerability reflected."""
+    return """
+    <html>
+        <body>
+            <input type="hidden" name="auth_handle" value="test123&quot;">
+            <input type="hidden" name="username" value="&quot;test123">
+        </body>
+    </html>
+    """
+
+
+def testCVE20142120_whenVulnerable_reportFinding(
+    requests_mock: req_mock.mocker.Mocker,
+) -> None:
+    """Unit test for CVE-2014-2120: case when target is vulnerable."""
+    # Mock the vulnerable response
```

Comment:
I don't think this and other comments are necessary.

Reply:
done